src/Components/DrillPage.js

```diff
@@ -1,22 +1,41 @@
-import React, { Component } from 'react';
-import { ScrollView, StyleSheet, View, Text, Button, ImageBackground, TouchableOpacity } from 'react-native';
+import React from 'react';
+import { ScrollView, StyleSheet, View, Text, Image, ImageBackground, TouchableOpacity } from 'react-native';
 import { connect } from 'react-redux';
+import ic_favorite_border from '../Images/ic_favorite_border.png';
+import ic_favorite from '../Images/ic_favorite.png';
 import theme from '../styles/theme.style';
 
-class DrillPage extends Component {
-  _toggleFavorite() {
-    const action = { type: 'TOGGLE_FAVORITE', value: this.state.dr };
-    this.props.dispatch(action);
-  }
-
-  componentDidUpdate() {
-    console.log('componentDidUpdate : ');
-    console.log(this.props.favoritesDrill);
+class DrillPage extends React.Component {
+  state = {
+    drill: this.props.drill,
+  };
+  _displayFavoriteImage() {
+    var sourceImage = ic_favorite_border;
+    if (this.props.favoritesDrill.findIndex(item => item.id === this.props.route.params.drill.id) !== -1) {
+      sourceImage = ic_favorite;
+    }
+    return <Image style={styles.favoriteImage} source={sourceImage} />;
   }
 
   render() {
     const drill = this.props.route.params.drill;
     const { navigation } = this.props;
+    const {
+      title,
+      infoWrapper,
+      DrillPage,
```

Comment:
You need to use the correct convention, in camelCase, so `drillPage` instead of `DrillPage`